src/Components/Navbar/Navbar.jsx

```diff
@@ -1,6 +1,121 @@
+import { useState, useEffect, useRef } from "react";
+import { NavLink } from "react-router-dom";
 import style from "./Navbar.module.scss";
+
 const Navbar = () => {
-  return <nav className={style.navbar}>Navbar</nav>;
+  const [burger, setBurger] = useState(false);
+
+  const toggleMenu = () => {
+    setBurger(!burger);
+  };
+  const handleKeyDown = (e) => {
+    if (e.key === "Enter") {
+      toggleMenu();
+    }
+  };
+
+  const menuRef = useRef();
+  const hamRef = useRef();
+
+  useEffect(() => {
+    const handler = (e) => {
+      if (!menuRef.current.contains(e.target) && !hamRef.current.contains(e.target)) {
+        setBurger(false);
+      }
+    };
+    document.addEventListener("mousedown", handler);
+    return () => {
+      document.removeEventListener("mousedown", handler);
```

Comment:
I see you have added listeners to the document element here to detect clicks outside of the menu but try to figure out another way where you don't have to use the `addEventListener` function as it is not recommended to handle events this way.